src/components/StepIndicator/StepIndicator.tsx

```diff
@@ -0,0 +1,93 @@
+
+type Step = { label: string; order: number };
+
+interface Props extends IntrinsicElements {
+  currentProgress: number;
+  steps: Step[];
+  counters?: boolean;
+  smallCounters?: boolean;
+  showLabels?: boolean;
+  isCentered?: boolean;
+}
+
+/**
+ * **StepIndicator Component**
+ *
+ * @param {number}    currentProgress    Number tracker of which step the user is currently on.
+ * @param {Step[]}    steps              An array of step object data which consist of a label and the order number. The order number is 1 based.
+ * @param {boolean}   counters           Boolean which shows the counters alternative to the step indidcator.
+ * @param {boolean}   smallCounters      Boolean which shows the smaller counters alternative to the step indicator. Only shows if counters is true.
+ * @param {boolean}   showLabels         Dictates whether or not labels are show under each step.
+ * @param {boolean}   isCentered         Dictates whether the labels are centered under the steps. Labels must be true.
+ */
+export const StepIndicator: React.FC<Props> = ({
+  counters = false,
+  smallCounters = false,
+  showLabels = true,
+  currentProgress,
+  steps,
+  isCentered = false,
+  ...rest
+}) => {
+  return (
+    <div
+      className={`usa-step-indicator${
+        isCentered ? " usa-step-indicator--center" : ""
+      }${
+        counters
+          ? ` usa-step-indicator--counters${smallCounters ? "-sm" : ""}`
+          : ""
+      }`}
+      aria-label="progress"
+      role="progressbar"
+      {...rest}
+    >
+      <ol className="usa-step-indicator__segments">
+        {steps.map((step) => {
+          return (
+            <li
+              key={step.label}
+              className={`usa-step-indicator__segment ${
+                step.order === currentProgress
+                  ? "usa-step-indicator__segment--current"
+                  : step.order < currentProgress
+                  ? "usa-step-indicator__segment--complete"
+                  : ""
+              }`}
+            >
+              <span className="usa-step-indicator__segment-label">
+                {showLabels && step.label}
+                {step.order !== currentProgress && (
+                  <span className="usa-sr-only">
+                    {step.order > currentProgress
+                      ? "completed"
+                      : "not completed"}
+                  </span>
+                )}
+              </span>
+            </li>
+          );
+        })}
+      </ol>
+      <div className="usa-step-indicator__header">
+        <h4 className="usa-step-indicator__heading">
+          <span className="usa-step-indicator__heading-counter">
+            <span className="usa-sr-only">Step</span>
+            <span className="usa-step-indicator__current-step margin-right-05">
+              {currentProgress}
+            </span>
+            <span className="usa-step-indicator__total-steps">
+              of {steps.length}
+            </span>{" "}
```

Comment:
Kindof a question for HCD, but should there be a "done" state? Currently when you're on 5/5 the last step is the lighter blue. Should there be a way to visually indicate this is complete? I tried to illustrate this by setting `currentProgress` to 6.

Step 5:
<img width="1204" alt="Screen Shot 2022-09-29 at 3 04 36 PM" src="https://user-images.githubusercontent.com/22647707/193121389-6a15d3a0-e149-4f66-aa37-46608c3e7b8a.png">

Step 6:
<img width="1167" alt="Screen Shot 2022-09-29 at 3 04 45 PM" src="https://user-images.githubusercontent.com/22647707/193121406-4e6b5ab6-12db-4a23-aea1-40c0e631ea3e.png">